Dockerfile

```diff
@@ -239,10 +228,25 @@ COPY --from=hashicorp/packer:1.4.2 /bin/packer /bin/packer
 COPY hack/installer/packer.json /packer.json
 COPY hack/installer/entrypoint.sh /bin/entrypoint.sh
 COPY --from=kernel /vmlinuz /usr/install/vmlinuz
-COPY --from=initramfs-build /usr/lib/syslinux/ /usr/lib/syslinux
+COPY --from=initramfs-base /usr/lib/syslinux/ /usr/lib/syslinux
 COPY --from=initramfs /initramfs.xz /usr/install/initramfs.xz
 COPY --from=rootfs /rootfs.tar.gz /usr/install/rootfs.tar.gz
-COPY --from=osctl-linux-amd64-build /osctl-linux-amd64 /bin/osctl
+COPY --from=osctl-linux-build /osctl-linux-amd64 /bin/osctl
 ARG TAG
 ENV VERSION ${TAG}
 ENTRYPOINT ["entrypoint.sh"]
+
+# The test target performs tests on the source code.
+
+FROM base AS test
+RUN ["/toolchain/bin/unlink", "/etc/ssl"]
+COPY --from=rootfs-base / /
+RUN ["/toolchain/bin/ln", "-svf", "/toolchain/bin/bash", "/bin/sh"]
+COPY hack/golang/test.sh /bin
+
+# The lint target performs linting on the source code.
+
+FROM base AS lint
+RUN curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b /toolchain/bin v1.16.0
```

Comment:
maybe as a todo for later is move this step to bldr/pkgs so we can skip out on the install each time